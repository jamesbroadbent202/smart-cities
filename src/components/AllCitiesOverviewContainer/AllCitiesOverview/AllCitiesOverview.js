import React from 'react';
import PropTypes from 'prop-types';
import PageWrapper from '../../PageWrapper/PageWrapper';
import CategoryOverviews from '../../CategoryOverviews/CategoryOverviews';
import Icon from '../../Icon/Icon';
import { LINKS } from '../../../constants';
import taylorAngus from './taylorAngus.png';
import COLORS from '../../../style/_colors.scss';
import style from './AllCitiesOverview.scss';

const AllCitiesOverview = props => (
  <PageWrapper>
    <div className={style.heroWrapper}>
      <p className={style.heroText}>
        Creating the foundations for success across
        all <span className={style.heroTextCityCount}>{props.cities.length}</span> cities
      </p>
    </div>

    <div className={style.infoWrapper}>
      <div className={style.infoContainer}>
        <div className={style.infoTextWrapper}>
          <h1 className={style.infoTitle}>About</h1>

          <p className={style.infoBodyText}>
            The Australian Government is committed to the continual improvement of our cities.
            Governments need to track and compare cities’ performance in order to
            make the best policy and investment decisions for Australia’s future.
          </p>

          <p className={style.infoBodyText}>
            <a
              className={style.cpnLink}
              href={LINKS.CITIES_PERFORMANCE_NETWORK}
              target="_blank"
              rel="noopener"
            >
              The National Cities Performance Framework
            </a> will support greater awareness and understanding of Australia’s cities
            among policy makers and interested Australians.
          </p>
        </div>

        <figure className={style.ministerPhotoWrapper}>
          <div className={style.ministerPhotoClip}>
            <img
              className={style.ministerPhoto}
              src={taylorAngus}
              alt="The Honorable Angus Taylor MP"
            />
          </div>

          <figcaption>
            <p className={style.ministerTitle}>The Hon Angus Taylor MP</p>

            <p>Assistant Minister for Cities & Digital Transformation</p>
          </figcaption>
        </figure>
      </div>
    </div>

    <div className={style.subInfoContainer}>
      <div className={style.subInfoTextWrapper}>
        <h1 className={style.infoTitle}>Policy priorities & context</h1>

        <p className={style.subInfoBodyText}>
          The National Cities Performance Framework is designed to measure how well
          our cities are performing against the Australian Government’s 6 Smart Cities
          policy priorities areas:
        </p>
      </div>

      <Icon
        className={style.subInfoIcon}
        icon="city"
        size={120}
        color={COLORS.PRIMARY_600}
      />
    </div>

    <CategoryOverviews cities={props.cities} />
  </PageWrapper>
);

AllCitiesOverview.propTypes = {
  cities: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    indices: PropTypes.object.isRequired,
  })).isRequired,
};

export default AllCitiesOverview;
