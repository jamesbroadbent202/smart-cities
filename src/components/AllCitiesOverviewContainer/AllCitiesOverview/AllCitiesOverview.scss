@import '~style/variables';

$minister-image-width: 214px;

.heroWrapper {
  display: flex;
  align-items: flex-end;
  height: 350px;
  // this needs to sit up behind the category nav
  margin-top: -$top-nav-height-mobile;
  background: url('./heroImage.jpg') center/cover no-repeat;

  @include for-tablet-portrait-up {
    height: 450px;
  }

  @include for-desktop-up {
    height: 550px;
    margin-top: -$top-nav-height-desktop;
  }
}

.heroText {
  @include container;

  width: 100%;
  margin-bottom: 32px;
  line-height: 1.2;
  font-size: 27px;
  font-weight: 300;
  letter-spacing: 1px;
  color: $color-white;

  @include for-tablet-portrait-up {
    padding-right: 20%;
    margin-bottom: 40px;
    font-size: 36px;
    letter-spacing: 0;
  }

  @include for-tablet-landscape-up {
    margin-bottom: 50px;
    font-size: 46px;
  }

  @include for-desktop-up {
    margin-bottom: 60px;
    font-size: 66px;
    letter-spacing: -2.6px;
  }
}

.heroTextCityCount {
  font-weight: 700;
}

.infoWrapper {
  background: $color-primary--600;
  color: $color-white;
}

.infoContainer {
  @include container;

  display: flex;
  flex-direction: column;
  padding-top: 42px;
  padding-bottom: 32px;

  @include for-tablet-landscape-up {
    flex-direction: row;
    padding-bottom: 118px;
  }
}

.infoTextWrapper {
  margin-bottom: 32px;

  @include for-tablet-landscape-up {
    margin-right: 32px;
    margin-bottom: 0;
  }
}

.ministerPhotoWrapper {
  color: $color-white;
  font-size: 14px;
  text-align: center;
  align-self: center;

  @include for-tablet-landscape-up {
    flex: none;
    width: $minister-image-width;
  }
}

.ministerPhotoClip {
  @include shadow;

  border-radius: 100%;
  width: $minister-image-width;
  height: $minister-image-width;
  margin: 0 auto 16px;
  overflow: hidden;
}

.ministerPhoto {
  width: 100%;
  height: 100%;
  transition: transform $animation-duration;
  transform-origin: 50% 50%;

  &:hover {
    transform: scale(1.05);
  }
}

.ministerTitle {
  font-weight: 700;
}

.subInfoContainer {
  @include container;

  padding-top: 42px;
  padding-bottom: 42px;
  display: flex;
  flex-direction: column;

  @include for-tablet-portrait-up {
    flex-direction: row;
  }
}

.subInfoTextWrapper {
  flex: auto;
  margin-bottom: 32px;

  @include for-tablet-portrait-up {
    margin-bottom: 0;
    margin-right: 64px;
  }
}

.subInfoIcon {
  margin: 0 auto;

  @include for-tablet-portrait-up {
    margin: 14px 0 0; // visually align the top with the title text
    flex: none;
    width: 140px;
  }
}

.infoTitle,
.subInfoTitle {
  font-size: 44px;
  margin-bottom: 32px;
  line-height: 1.2;
  letter-spacing: -1.8px;
}

.subInfoBodyText,
.infoBodyText {
  font-size: 22px;
  line-height: 37px;
  letter-spacing: -0.7px;
}

.infoBodyText:not(:first-of-type) {
  padding-top: 38px;
}

.cpnLink {
  text-decoration: $color-highlight underline;
}
