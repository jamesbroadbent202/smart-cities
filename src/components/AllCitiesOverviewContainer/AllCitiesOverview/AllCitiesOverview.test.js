import React from 'react';
import { shallow } from 'enzyme';
import AllCitiesOverview from './AllCitiesOverview';

jest.mock('../../CategoryOverviews/CategoryOverviews', () => 'CategoryOverviews');
jest.mock('../../Icon/Icon', () => 'Icon');

const defaultProps = {
  cities: [
    {
      id: 'perth',
      indices: {},
    },
    {
      id: 'sydney',
      indices: {},
    },
  ],
};

it('should match Snapshot', () => {
  const component = shallow(
    <AllCitiesOverview {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});
