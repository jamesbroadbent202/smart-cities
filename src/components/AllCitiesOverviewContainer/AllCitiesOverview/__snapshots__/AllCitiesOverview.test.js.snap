// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<PageWrapper categoryId=\\"overview\\" cityId=\\"all-cities\\">
  <div className=\\"heroWrapper\\">
    <p className=\\"heroText\\">
      Creating the foundations for success across all 
      <span className=\\"heroTextCityCount\\">
        2
      </span>
       cities
    </p>
  </div>
  <div className=\\"infoWrapper\\">
    <div className=\\"infoContainer\\">
      <div className=\\"infoTextWrapper\\">
        <h1 className=\\"infoTitle\\">
          About
        </h1>
        <p className=\\"infoBodyText\\">
          The Australian Government is committed to the continual improvement of our cities. Governments need to track and compare cities’ performance in order to make the best policy and investment decisions for Australia’s future.
        </p>
        <p className=\\"infoBodyText\\">
          <a className=\\"cpnLink\\" href=\\"https://cities.dpmc.gov.au/\\" target=\\"_blank\\" rel=\\"noopener\\">
            The National Cities Performance Framework
          </a>
           will support greater awareness and understanding of Australia’s cities among policy makers and interested Australians.
        </p>
      </div>
      <figure className=\\"ministerPhotoWrapper\\">
        <div className=\\"ministerPhotoClip\\">
          <img className=\\"ministerPhoto\\" src=\\"taylorAngus.png\\" alt=\\"The Honorable Angus Taylor MP\\" />
        </div>
        <figcaption>
          <p className=\\"ministerTitle\\">
            The Hon Angus Taylor MP
          </p>
          <p>
            Assistant Minister for Cities &amp; Digital Transformation
          </p>
        </figcaption>
      </figure>
    </div>
  </div>
  <div className=\\"subInfoContainer\\">
    <div className=\\"subInfoTextWrapper\\">
      <h1 className=\\"infoTitle\\">
        Policy priorities &amp; context
      </h1>
      <p className=\\"subInfoBodyText\\">
        The National Cities Performance Framework is designed to measure how well our cities are performing against the Australian Government’s 6 Smart Cities policy priorities areas:
      </p>
    </div>
    <Icon className=\\"subInfoIcon\\" icon=\\"city\\" size={120} color=\\"PRIMARY_600\\" />
  </div>
  <CategoryOverviews cities={{...}} />
</PageWrapper>"
`;
